src/futils.cpp

```diff
@@ -58,9 +59,13 @@ namespace Exiv2 {
     const char* ENVARKEY[] = {"EXIV2_HTTP_POST", "EXIV2_TIMEOUT"}; //!< @brief request keys for http exiv2 handler and time-out
 // *****************************************************************************
 // free functions
-    std::string getEnv(EnVar var) {
+    std::string getEnv(EnVar var)
+    {
+        if (var < envHTTPPOST || var > envTIMEOUT) {
+            throw std::range_error("Unexpected env variable");
```

Comment:
I think most of the exiv2 code throws only instances of `Exiv2::Error`. To avoid bad surprises I would keep it that way rather than throwing an exception from std, although it might be better suited here.

Reply:
True, I did not think about that. I will change it

Reply:
Hummm, I have been checking this out and I do not like much the idea of using `Exiv2::Error` here. The constructor of this class always takes an integer, something that does not make sense here (Unless it already exists a table of values, and some of them could be used for cases like this one.) Otherwise I think that and std exception like `std::range_error` is better suited here.

Reply:
I thought that @D4N was going to add an enum kerBlaDeeBla .... to give symbol names the error.  That's what we should be throwing and I believe they will be localised.

Reply:
I understand but if that enum is not available yet ... what should I do ? Would be this enough?

```
if (var < envHTTPPOST || var > envTIMEOUT) {
    const int REPLACE_BY_ENUM_VALUE=123;
    throw Error(REPLACE_BY_ENUM_VALUE);
}
```
So that we can localise these values once the enum is ready ?

Reply:
I didn't have time to work on #109, so the enums are unfortunately not yet available.

How about we solve this issue in the following way: we replace the variables `ENVARDEF` and `ENVARKEY` with a `std::vector/array<std::string>`. Thereby, we do not need to perform a range check on the enum, as std::vector will do that itself (provided we use `.at()` in the `getEnv` function). Then throwing a `range_error` is imho ok here, as you really have to force it as a library user, provided that we document it.

Reply:
This sounds like a perfect good solution to me :). Then, instead of throwing `range_error` , the std::vector::at() method throws with `out_of_range`:

http://www.cplusplus.com/reference/vector/vector/at/

I will commit the change in few minutes